### ISSUE TITLE
Bot crashes when a group is changed into a supergroup

### ISSUE THREAD

**Issue by noiob** [telegram-rs/telegram-bot#36]
`An error occured: JsonDecode(ApplicationError("No field for inferring message type is set"))`

**Comment by noiob** [comment]
Okay, so I've found a message from a group being changed into a supergroup:
`"message":{"message_id":28164,"from":{"id":28111482,"first_name":"redacted","last_name":"redacted","username":"redacted"},"chat":{"id":-137980893,"title":"test group","type":"group"},"date":1458925136,"migrate_to_chat_id":-1001026227392}},{"update_id":591487127,
`

So the problem most likely is imho in the function `fn decode<D: Decoder>(d: &mut D) -> Result<Self, D::Error>` [(line 410 in `mod.rs`)](https://github.com/LukasKalbertodt/telegram-bot/blob/master/src/types/mod.rs#L410):
The only check for supergroup migration (I haven't found a way of directly creating a supergroup in any of the clients I use, btw) is this:
```
if let Some(true) = try!(d.read_struct_field(
            "supergroup_chat_created", 0, Decodable::decode)) {
            return Ok(MessageType::SuperGroupChatCreated(GroupToSuperGroupMigration {
                from: try_field!(d, "migrate_from_chat_id"),
                to: try_field!(d, "migrate_to_chat_id"),
            }))
        };
```
However, the message only contains `migrate_to_chat_id`, which in turn causes an error.

[This is propably too long and not the right format for an issue, so apologies in advance, thanks for this awesome library!]

**Comment by knsd** [comment]
Thanks, @noiob!

I will fix it today.

**Comment by knsd** [comment]
Fixed in #38 and in upstream.
